02-peer-protocol.md

```diff
@@ -566,12 +579,18 @@ The funding node:
     - SHOULD send a `closing_signed` message.
 
 The sending node:
-  - MUST set `fee_satoshis` less than or equal to the
- base fee of the final commitment transaction, as calculated in [BOLT #3](03-transactions.md#fee-calculation).
-  - SHOULD set the initial `fee_satoshis` according to its
- estimate of cost of inclusion in a block.
-  - MUST set `signature` to the Bitcoin signature of the close
- transaction, as specified in [BOLT #3](03-transactions.md#closing-transaction).
+  - If the channel does not use `option_anchor_outputs`:
+    - MUST set `fee_satoshis` less than or equal to the base fee of the final
+    commitment transaction, as calculated in [BOLT #3](03-transactions.md#fee-calculation).
```

Comment:
Our commitment transaction, or our counterparty's commitment transaction (I know this isn't new on this PR, I just realized it).

Reply:
That's a good point, it's a subtle point. We use our commitment transaction in eclair, and I think it's what makes most sense: we must create a transaction that's cheaper than the alternative we can unilaterally broadcast (even though you'd also need to factor in csv delays into the choice of which one to broadcast).

However, the receiver would probably want to receive a proposal that also pays less fees than *his* commit tx, so would it make more sense to use the min of both commit tx fees here?

Reply:
Given the recipient MUST fail/force-close if we're too high, min seems like the obvious choice. Are there any edge-cases where one party has a high dust limit, causing one side of the commitment transaction to be missing an output (skewing the fee pretty low) whereas both outputs would appear in the cooperative shutdown case? In that world you risk always having to force-close (but maybe that's what you want - at least if the side that gets to skip the counterparty's output is the initiator who pays fees).

Reply:
I don't think this edge case can meaningfully happen thanks to the channel reserve (which must be greater than the `dust_limit`), unless the channel has just been opened and the reserve requirement was never met.

I'm happy to change this to specify that it should be the min of both commitments (even though this requirement will disappear anyway since it doesn't apply to anchor outputs channels).

Reply:
Actually, should we drop this requirement for the channel initiator? I don't see a reason why the node not paying the fee should care one bit if the channel opener wants to close with a higher feerate than the existing channel one.

Reply:
That's what I did in the first version of the PR, I had dropped this requirement completely, but in order to keep existing channel behavior I was asked to drop it only for anchor outputs. I'm happy to remove it for all channel types honestly, probably worth discussing again during the next spec meeting to see if we can convince @rustyrussell ;)

Reply:
Right, its a compatibility issue with existing nodes, so not ideal, but its not a super critical one imo - a few extra force-closes instead of cooperative closes while nodes upgrade isn't the end of the world. If it resulted in force-closes of otherwise-functional channels it'd be much worse, but we're talking about closing the channel anyway.